tests/common-tests.h

```diff
@@ -55,7 +55,20 @@ inline namespace boost {
 #endif
 #endif // HAVE_BOOST
 
-
+// Firebird returns string that represents entire byte-size(not character size) of the column( i.e. CHAR(10) NONE - 10bytes, CHAR(10) UTF8 - 40bytes )
+// MySql returns right trimmed value by default or space padded up to column character size when PAD_CHAR_TO_FULL_LENGTH
+// PostgreSQL returns space padded version up to actual column character size (not byte size )
+// SQLite has TEXT affinity for all char types so it is not an issue here - macro bellow will match correctly
+// TODO: comments for other backends(DB, ODBC)
+//This macro will check result string from column defined as fixed char
+//It will check only bytes up to original string size.
+//If result string is bigger than original string then all remaining chars must be spaces so if any non-space char is found it will fail .
+#define CHECK_CHAR( STR, ORIG_STR ) \
+    CHECK(strncmp(std::string(STR).c_str(),std::string(ORIG_STR).c_str(),std::string(ORIG_STR).size()) == 0); \
```

Comment:
The trouble is that if we use a function, the location of (failing) `CHECK`s inside it won't be correct as it would use the line number corresponding to the code of the function itself instead of the code calling it, this is the only reason to use macros for checks like this.

Unfortunately I don't think Catch provides macro like `CHECK_WITH_LOCATION(..., file, line)` that we'd need to implement such a function properly.